sdk/identity/azure-identity/migration_guide.md

```diff
@@ -0,0 +1,38 @@
+# Guide for migrating to azure-identity from azure-common
+
+## JSON- and file-based authentication
+
+To encourage best security practices, `azure-identity` does not support JSON- and file-based authentication in the same
+way as `azure-common`. `azure-common` provided factory methods like [`get_client_from_json_dict`][client_from_json] and
+[`get_client_from_auth_file`][client_from_auth_file] to provide credentials as plaintext JSON. Azure credentials, like
+passwords, should not be stored in plaintext whenever possible. You can find information about Azure Active Directory
+(Azure AD) and best practices for authentication in [Azure AD's documentation][azure_ad].
+
+If it's too difficult to migrate away from JSON- or file-based authentication despite the security risks, you can still
+use `azure-identity` credentials. With a JSON file containing the credentials, you can use [`json.load`][json] to
+authenticate a service principal with a [`ClientSecretCredential`][client_secret_cred]:
+```python
+import json
+from azure.identity import ClientSecretCredential
+from azure.keyvault.certificates import CertificateClient
+
+with open("credentials.json") as json_file:
+    json_dict = json.load(json_file)
+    
+credential = ClientSecretCredential(
+    tenant_id=json_dict["tenantId"],
+    client_id=json_dict["clientId"],
+    client_secret=json_dict["clientSecret"],
+    authority=json_dict["activeDirectoryEndpointUrl"]
+)
+client = CertificateClient("https://{vault-name}.vault.azure.net", credential)
```

Comment:
I think before and after code would help make plain the connection between this pattern and `get_client_from_auth_file(KeyVaultManagementClient, file_path)`.